---
title: "TCGA_analysis"
author: "John Lee"
date: "Oct 9, 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load Packages


```{r load,message=FALSE}
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(SummarizedExperiment)
```

## Query

```{r pressure}
query.exp.hg19 <- GDCquery(project = "TCGA-GBM",
                  data.category = "Gene expression",
                  data.type = "Gene expression quantification",
                  platform = "Illumina HiSeq", 
                  file.type  = "normalized_results",
                  experimental.strategy = "RNA-Seq",
                  barcode = c("TCGA-14-0736-02A-01R-2005-01", "TCGA-06-0211-02A-02R-2005-01"),
                  legacy = TRUE)
```

### Show HTML grid
```{r}
datatable(getResults(query.exp.hg19), 
              filter = 'top',
              options = list(scrollX = TRUE, keys = TRUE, pageLength = 5), 
              rownames = FALSE)
```  

## Download Data
```{r}
GDCdownload(query.exp.hg19, method = "api")
```  
### get the matrix
```{r}
data <- GDCprepare(query.exp.hg19)
datatable(as.data.frame(colData(data)), 
              options = list(scrollX = TRUE, keys = TRUE, pageLength = 5), 
              rownames = FALSE)
datamatrix <- assay(data)
```
